src/components/02_atoms/OpsModalButton/OpsModalButton.js

```diff
@@ -0,0 +1,83 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Button from '@material-ui/core/Button';
+import Dialog from '@material-ui/core/Dialog';
+import DialogActions from '@material-ui/core/DialogActions';
+import DialogContent from '@material-ui/core/DialogContent';
+import DialogContentText from '@material-ui/core/DialogContentText';
+import DialogTitle from '@material-ui/core/DialogTitle';
+import IconButton from '@material-ui/core/IconButton';
+import Slide from '@material-ui/core/Slide';
+
+function Transition(props) {
+  return <Slide direction="up" {...props} />;
+}
+
+/* A Drupal specific button, for use in the operations column of a table.
+ *
+ * Typically a Add, Edit or Delete icon can be provided as a child element
+ * see for example  '@material-ui/icons/Add'
+ *
+ * When pressed a modal dialog slides up into view.
+ *
+ * Custom Properties :-
+ *   title, string to appear in the modal dialog.
+ *   text, string to appear below the title.
+ */
+class OpsModalButton extends React.Component {
+  state = {
+    open: false,
+  };
+
+  handleClickOpen = () => {
+    this.setState({ open: true });
+  };
+
+  handleClose = () => {
+    this.setState({ open: false });
+  };
+
+  render() {
+    // Pass all props except those consumed here down into the button.
+    const { title, text, ...buttonProps } = this.props;
+
+    return (
+      <IconButton {...buttonProps} onClick={this.handleClickOpen}>
+        {this.props.children}
+        <Dialog
+          open={this.state.open}
+          TransitionComponent={Transition}
+          keepMounted="keepMounted"
+          onClose={this.handleClose}
+          aria-labelledby="alert-dialog-slide-title"
```

Comment:
I am wondering what happens if something reuses this component? Wouldn't this lead to duplicate IDs? One for each row for example? Maybe we need to allow to pass in a prefix? ❓

Reply:
Thanks for highlighting potential problems. It is really helpful -- that section of the code just some sample code.... copied from the material-ui website so I may have to component-ize it.

It is on my todo list ... that is I don't know the correct response yet... but I will work out an informed response by the next iteration.